### PR TITLE
Update maven plugin to 4.0.1

### PR DESCRIPTION
**Pull request by zlav**
# Overview

This PR updates the maven plugin to version 4.0.1

## Acceptance criteria

The issues related to newer Maven projects that we have seen are resolved. A common issue we have seen is
```
A required class was missing while executing com.github.ferstl:depgraph-maven-plugin:3.3.0:aggregate: org/apache/commons/lang3/StringUtils
```

## Testing plan

_How did you validate that this PR works? What literal steps did you take when manually checking that your code works?_

_Example:_

1. _Set up test case X._
2. _Run command Y. Make sure Z happens._

_This section should list concrete steps that a reviewer can sanity check and repeat on their own machine (and provide any needed test cases)._

## Risks

JDK 7 users won't be able to use the plugin

## References

- _[PS-9](https://fossa.atlassian.net/browse/PS-9): Implement X._

## Checklist

- [ ] I added tests for this PR's change (or explained in the PR description why tests don't make sense).
- [ ] If this PR introduced a user-visible change, I added documentation into `docs/`.
- [ ] If this change is externally visible, I updated `Changelog.md`. If this PR did not mark a release, I added my changes into an `# Unreleased` section at the top.
- [ ] If I made changes to `.fossa.yml` or `fossa-deps.{json.yml}`, I updated `docs/references/files/*.schema.json`. You may also need to update these if you have added/removed new dependency type (e.g. `pip`) or analysis target type (e.g. `poetry`).

### CODE REVIEW

**Comment by cmboling** [comment]
Thanks for setting this up! Taking a look today 💯

**Comment by meghfossa** [comment]
Can you also update versioning information in https://github.com/fossas/fossa-cli/blob/master/docs/references/strategies/languages/maven/mavenplugin.md#maven-plugin?

File: src/Strategy/Maven.hs [hunk 1]
```diff
@@ -100,6 +100,16 @@ getDepsPlugin ::
...
 
+getDepsPluginLegacy ::
+  ( Has (Lift IO) sig m
+  , Has Diagnostics sig m
+  , Has ReadFS sig m
+  , Has Exec sig m
+  ) =>
+  MavenProjectClosure ->
+  m (Graphing Dependency, GraphBreadth)
+getDepsPluginLegacy closure = context "Plugin analysis" (Plugin.analyzeLegacy' . parent $ PomClosure.closurePath closure)
```
Comment (meghfossa) [inline src/Strategy/Maven.hs:111]: wording: Legacy Plugin Analysis

**Review by meghfossa — APPROVED** [review]
LGTM. This feels like a release (make sure to update the changelog). I did not perform e/e testing, as the code looked straightforward - let me know if you want me to execute any specific testplan. 

Builds are failing in osx: you will need to rebase after this gets merged: https://github.com/fossas/fossa-cli/pull/924

**Pull request merged**